### PR TITLE
Fix set the replicationClusters param

### PR DESCRIPTION
**Pull request by nodece**
There are should get the  `ReplicationClusters ` from the message body, otherwise always `null`.

### CODE REVIEW

**Comment by Lanayx** [comment]
Hi, thank you for the PR, but the build is failing. Can you also remove[ this property](https://github.com/fsharplang-ru/pulsar-client-dotnet/blob/develop/src/Pulsar.Client/Internal/BatchMessageContainer.fs#L107) as it will be unused

File: src/Pulsar.Client/Internal/BatchMessageContainer.fs [hunk 1]
```diff
@@ -128,7 +128,7 @@ type internal DefaultBatchMessageContainer<'T>(prefix: string, config: ProducerC
             PartitionKey = batchItems.[0].Message.Key
             OrderingKey = batchItems.[0].Message.OrderingKey
             TxnId = this.CurrentTxnId
-            ReplicationClusters = this.ReplicationClusters
+            ReplicationClusters = batchItems.[0].ReplicationClusters
```
Comment (RobertIndie) [inline src/Pulsar.Client/Internal/BatchMessageContainer.fs]: This seems to work only for the replication clusters of the first message. If the first message is specified to be replicated to cluster A, then the second message is specified to be replicated to cluster B. The entire batch of messages does not appear to be replicated to cluster B, and this behavior should not be what we expect. Please correct me if I am wrong.

**Review by RobertIndie — COMMENTED** [review]
Could you add some tests to verify the changes?

File: src/Pulsar.Client/Internal/BatchMessageContainer.fs (hunk 1 above)
Comment (nodece) [reply src/Pulsar.Client/Internal/BatchMessageContainer.fs]: The `batchItems` should be a queue, and have many messages, we will remove the first element from this queue after calling `CreateOpSendMsg`, so it should be working.

Comment (Lanayx) [reply src/Pulsar.Client/Internal/BatchMessageContainer.fs]: @RobertIndie as far as I can see in the java code it behaves exactly the same - replication settings are just taken from the first message in a batch

Comment (RobertIndie) [reply src/Pulsar.Client/Internal/BatchMessageContainer.fs]: @Lanayx In the java client, the replication clusters of the whole batch message are take from all single messages in batch. [Here is the code](https://github.com/apache/pulsar/blob/877bf3a34e72336c05706ffd48826e4347606196/pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L1614-L1616) But in the java client, this behavior should also be incorrect.
We need to get the messages of the same replication clusters into the same batch. @nodece Could we also fix this issue in this PR? I can give a fix on the java client side.

Comment (nodece) [reply src/Pulsar.Client/Internal/BatchMessageContainer.fs]: I can fix this issue on next PR.

Comment (Lanayx) [reply src/Pulsar.Client/Internal/BatchMessageContainer.fs]: @RobertIndie that's right, the code that you sent is only called for the first message of the batch, so all other messages in the batch will inherit that settings. I agree with @nodece  that we can fix the behavior in the next PR when you are ready with Java code changes.

**Review by Lanayx — APPROVED** [review]

**Pull request merged**